extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputReader.java

```diff
@@ -31,63 +31,135 @@
 import org.apache.druid.java.util.common.Pair;
 import org.apache.druid.java.util.common.logger.Logger;
 import org.apache.druid.java.util.common.parsers.CloseableIterator;
+import org.apache.druid.java.util.common.parsers.ParseException;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.NoSuchElementException;
 
 public class KafkaInputReader implements InputEntityReader
 {
   private static final Logger log = new Logger(KafkaInputReader.class);
-  private static final String DEFAULT_KEY_STRING = "key";
-  private static final String DEFAULT_TIMESTAMP_STRING = "timestamp";
 
   private final InputRowSchema inputRowSchema;
   private final KafkaRecordEntity record;
   private final KafkaHeaderReader headerParser;
   private final InputEntityReader keyParser;
   private final InputEntityReader valueParser;
-  private final String keyLabelPrefix;
-  private final String recordTimestampLabelPrefix;
-
+  private final String keyColumnPrefix;
+  private final String recordTimestampColumnPrefix;
+
+  /**
+   *
+   * @param inputRowSchema Actual schema from the ingestion spec
+   * @param record kafka record containing header, key & value
+   * @param headerParser Header parser for parsing the header section, kafkaInputFormat allows users to skip header parsing section and hence an be null
+   * @param keyParser Key parser for key section, can be null as well
+   * @param valueParser Value parser is a required section in kafkaInputFormat, but because of tombstone records we can have a null parser here.
+   * @param keyColumnPrefix Default key column prefix
+   * @param recordTimestampColumnPrefix Default kafka record's timestamp prefix
+   */
   public KafkaInputReader(
       InputRowSchema inputRowSchema,
       KafkaRecordEntity record,
       KafkaHeaderReader headerParser,
       InputEntityReader keyParser,
       InputEntityReader valueParser,
-      String keyLabelPrefix,
-      String recordTimestampLabelPrefix
+      String keyColumnPrefix,
+      String recordTimestampColumnPrefix
```

Comment:
those two fields are no longer prefixes, they are the actual column names, no?

Reply:
agree, done